src/snyk/common/languageserver/languageserver.ts

```diff
@@ -0,0 +1,98 @@
+import { getProxyEnvVariable, getProxyOptions } from '../proxy';
+import { ILanguageClientAdapter } from '../vscode/languageClient';
+import { ExtensionContext, LanguageClient, LanguageClientOptions, ServerOptions } from '../vscode/types';
+import { IVSCodeWorkspace } from '../vscode/workspace';
+
+export interface ILanguageServer {
+  start(): Promise<void>;
+
+  stop(): Promise<void>;
+
+  setConfig(config: IConfiguration): void;
+}
+
+export class LanguageServer implements ILanguageServer {
+  private context: ExtensionContext;
+  private languageClientAdapter: ILanguageClientAdapter;
+  private readonly workspace: IVSCodeWorkspace;
+
+  constructor(
+    context: ExtensionContext,
+    configuration: IConfiguration,
+    languageClientAdapter: ILanguageClientAdapter,
+    workspace: IVSCodeWorkspace,
+  ) {
+    this.configuration = configuration;
+    this.context = context;
+    this.languageClientAdapter = languageClientAdapter;
+    this.workspace = workspace;
+  }
+
+  private configuration: IConfiguration;
+  private client: LanguageClient;
+
+  setConfig(config: IConfiguration): void {
+    this.configuration = config;
+  }
+
+  async start(): Promise<void> {
+    // TODO remove feature flag when ready
+    if (!this.configuration.getPreviewFeatures().lsAuthenticate) {
+      return Promise.resolve(undefined);
+    }
+
+    // proxy settings
+    const proxyOptions = getProxyOptions(this.workspace);
+    const proxyEnvVariable = getProxyEnvVariable(proxyOptions);
+
+    // until a path can be configured
+    const serverOptions: ServerOptions = {
+      command: this.configuration.getSnykLanguageServerPath(),
+      args: ['-l', 'info'], // TODO file logging?
+      options: {
+        env: {
+          ...process.env,
+          // eslint-disable-next-line camelcase
+          https_proxy: proxyEnvVariable,
+          // eslint-disable-next-line camelcase
+          http_proxy: proxyEnvVariable,
+        },
+      },
+    };
+
+    // Options to control the language client
+    const clientOptions: LanguageClientOptions = {
+      documentSelector: [{ scheme: 'file', language: '' }],
+      initializationOptions: await this.getInitializationOptions(),
+    };
+    // Create the language client and start the client.
+    this.client = this.languageClientAdapter.create('Snyk LS', SNYK_LANGUAGE_SERVER_NAME, serverOptions, clientOptions);
+
+    // Start the client. This will also launch the server
+    return this.client.start();
+  }
+
+  async getInitializationOptions() {
+    return {
+      activateSnykCode: 'false',
+      activateSnykOpenSource: 'false',
+      activateSnykIac: 'false',
+      enableTelemetry: `${this.configuration.shouldReportEvents}`,
+      sendErrorReports: `${this.configuration.shouldReportErrors}`,
+      integrationName: CLI_INTEGRATION_NAME,
+      integrationVersion: await Configuration.getVersion(),
+      token: await this.configuration.getToken(),
+      cliPath: this.configuration.getCustomCliPath(),
```

Comment:
This currently passes custom CLI path (if provided only). Since this function doesn't return where current extension CLI lies, will it lead to second CLI download on language server? If yes, how about we pass determined CLI path using [this](https://github.com/snyk/vscode-extension/blob/559119069861df7847e82b7aa69fa4bccea9e824/src/snyk/cli/cliExecutable.ts#L2) until we move CLI management logic out of vs code codebase?

Reply:
Oh, I would have expected it to always have the correct CLI path. I'll try to integrate the Path from cliExecutable.ts then.